### DISCUSSION TITLE
pytest.fail() does not fail on a dashboard

### DISCUSSION THREAD

**Discussion by webmoskal** [seleniumbase/SeleniumBase#1196] [General]
pytest.fail() fails for pytest and fails on report.html, but displayed as passed on a dashboard

**Comment by mdmintz** [comment]
@webmoskal I wasn't able to reproduce. Can you try using ``self.fail()`` instead and see if you get the correct results?

**Reply by webmoskal** [reply]
E     AttributeError: 'TestLogin' object has no attribute 'fail'

```
@pytest.mark.usefixtures("sb")
class TestLogin():
   def test_fail(self):
      self.fail()
```

**Reply by mdmintz** [reply]
Use ``sb.fail()`` when using the SeleniumBase fixture format:

```python
# "sb" pytest fixture test in a method with no class
def test_sb_fixture_with_no_class(sb):
    sb.fail()

# "sb" pytest fixture test in a method inside a class
class Test_SB_Fixture:
    def test_sb_fixture_inside_class(self, sb):
        sb.fail()
```
